### PR TITLE
Corrections to currency format information

### PR DESCRIPTION
**Pull request by mDuo13**
Some corrections I found while digging through the `rippled` code and writing my transaction serialization code:

- The bit counts for the standard currency code were off by 1 byte
- The issued currency amount's exponent is represented as an unsigned integer by adding 97 to it (the comments in the code [wrongly stated 142 instead of 97](https://github.com/ripple/rippled/pull/2790/).)

### CODE REVIEW

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (seelabs) [inline content/references/rippled-api/api-conventions/currency-formats.md:40]: This is only true for non-native amounts. XRP does not use exponents. For XRP, the max allowed native amount on the network is 100000000000000000, which is 57 bits. So those high three bits would occupy where this paragraph says is the exponent.

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (seelabs) [inline content/references/rippled-api/api-conventions/currency-formats.md:42]: This is only true for non-native amounts. For XRP, it's the remaining 62 bits, not 54 bits; Also, for XRP the canonical range will be 0 to 100000000000000000. BTW, this is only 57 bits, though I would still say "62 bits" in the docs.

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (seelabs) [inline content/references/rippled-api/api-conventions/currency-formats.md]: I'd probably strike the work "rare". We either support the format or we don't. A currency code of all zeros is XRP.

**Review by seelabs — COMMENTED** [review]
Left a couple comments. BTW, I'm not sure how to interpret the `uxf` files. I assume those are specifying formatting (?), so I did not review those files.

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (mDuo13) [reply content/references/rippled-api/api-conventions/currency-formats.md:40]: Yes, in context this is only describing issued currencies (this section starts on lines 29-34, "issued currency math"). The section describing XRP is above it.

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (mDuo13) [reply content/references/rippled-api/api-conventions/currency-formats.md:42]: Yeah, this section is specifically describing issued currencies.

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (mDuo13) [reply content/references/rippled-api/api-conventions/currency-formats.md]: It is rare that a field specifies a currency code for XRP, though. The only case I can think of is path steps. I was hoping to head off confusion about any time you specify an XRP amount that you don't actually provide a currency code there (because the "not XRP" bit of the value handles it).

**Comment by mDuo13** [comment]
Sorry, I should have clarified that the `.uxf` files are the source files for the diagrams and don't need to be reviewed. (In case you're curious, I use a tool called [umlet](http://www.umlet.com/) for simple diagrams.)

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (seelabs) [reply content/references/rippled-api/api-conventions/currency-formats.md]: Yeah, I see where you're comming from. OK, I'm fine with keeping "rare".

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (seelabs) [reply content/references/rippled-api/api-conventions/currency-formats.md:42]: Ah, I missed that context. Thanks for clarifying.

**Review by seelabs — APPROVED** [review]
:+1:

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (jhaaaa) [inline content/references/rippled-api/api-conventions/currency-formats.md:34]: Sorry - not really within the scope of the PR - but just curious about what we mean by "internally"? Does it mean that as a dev using the `rippled` API, you'll never encounter values in this format in the API's requests and responses, but we're telling you how `rippled` stores these values behind the scenes to show how it supports high precision?

**Review by jhaaaa — APPROVED** [review]
I appreciate the graphics to help explain the concepts. But I have to admit that I don't understand how this works quite yet and my questions are beyond the scope of this PR so I'll hold off on them. From a pure language review - it looks good to me!

File: content/references/rippled-api/api-conventions/currency-formats.md
Comment (mDuo13) [reply content/references/rippled-api/api-conventions/currency-formats.md:34]: The idea behind this paragraph is:

- To let people know how to match `rippled`'s math exactly when it comes to issued currency amounts, so they know what to expect or can predict the outcomes of transactions down to the lowest decimal place. Or, more likely, to convince them that it's not worth the effort because they'd have to implement a bunch of low-level number routines, but they can come reasonably close by using a "big decimal" library with 15 digits of precision.
- To inform those implementing transaction serialization/signing that the binary format fits all that stuff in 64 bits, which also serves as a justification for why the first point is the way it is.

Some of this stuff is a little extra low-level and should probably get migrated to the Serialization Format page eventually (sometime after #483 is merged).

Comment (jhaaaa) [reply content/references/rippled-api/api-conventions/currency-formats.md:34]: Thank you for taking the time to provide such thorough and thoughtful context for me. Got it. 🙌

**Comment by mDuo13** [comment]
Rebased onto master to get a broken link fix.

**Pull request merged**
